pkg/session/session.go

```diff
@@ -95,28 +127,53 @@ func GetOrCreate(
 	return &session, nil
 }
 
-func newClient(ctx context.Context, url *url.URL, thumprint string) (*govmomi.Client, error) {
+func newClient(ctx GetOrCreateContext, sessionKey string, url *url.URL, thumprint string, feature Feature) (*govmomi.Client, error) {
```

Comment:
~~If we are not using the `logger` here, we could just pass the context instead.~~